classes/task/send_files.php

```diff
@@ -94,33 +94,28 @@ public function plagiarism_vericite_send_files()
                     $reportMetaData['external_content_data'] = array(new \Swagger\Client\Model\ExternalContentData($externalContentData));
                     $apiArgs['report_meta_data'] = new \Swagger\Client\Model\ReportMetaData($reportMetaData);
                     $result = plagiarism_vericite_call_api($plagiarismsettings['vericite_api'], PLAGIARISM_VERICITE_ACTION_REPORTS_SUBMIT_REQUEST, $apiArgs);
-
                     if (!empty($result) && is_array($result)) {
                         foreach ($result AS $externalcontentuploadinfo) {
                             // Now see if there are any presigned URLs we need to upload our attachment to.
                             if ($externalcontentuploadinfo) {
                                 plagiarism_vericite_log("urlPost:\n" . $externalcontentuploadinfo->getUrlPost() . "\nfilepath: " . $filepath);
-                                $ch = curl_init();
-                                curl_setopt($ch, CURLOPT_URL, $externalcontentuploadinfo->getUrlPost());
-                                $fh_res = fopen($filepath, 'r');
-                                curl_setopt($ch, CURLOPT_INFILE, $fh_res);
-                                curl_setopt($ch, CURLOPT_INFILESIZE, filesize($filepath));
-                                curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 120);
-                                curl_setopt($ch, CURLOPT_PUT, 1);
-                                curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+                                $c = new curl();
+                                $url = $externalcontentuploadinfo->getUrlPost();
+                                $params = array('file' => $filepath);
+                                $options = array('CURLOPT_CONNECTTIMEOUT' => 120);
                                 //Set Encryption from header
                                 if ($externalcontentuploadinfo->getHeaders()) {
                                     $headers = [];
                                     foreach ($externalcontentuploadinfo->getHeaders() as $key => $value) {
                                         $headers[] = "$key: $value";
                                     }
-                                    curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
+                                    $options['CURLOPT_HTTPHEADER'] = $headers;
```

Comment:
@baholladay can you change this line to 

$c->setHeader($headers);

Reply:
Same error:

<Error><Code>InvalidArgument</Code><Message>Only one auth mechanism allowed; only the X-Amz-Algorithm query parameter, Signature query string parameter or the Authorization header should be specified</Message><ArgumentName>Authorization</ArgumentName><ArgumentValue>Basic YW5vbnltb3VzOiBub3JlcGx5QG1vb2RsZS5vcmc=</ArgumentValue><RequestId>B519BDE93F44994E</RequestId><HostId>f0v7eZpxUTbGkssWrCKxgSNih6lp5YnYCPkWKxKkK+PzlE27ybUZUc4MN9HSJ+Imz3hpdRziALk=</HostId></Error>